meson.build

```diff
@@ -0,0 +1,307 @@
+    ['getpagesize', '#include <unistd.h>'],
+]
+
+sizeof_types = [
+    'char',
+    'int',
+    'long',
+    'long long',
+    'short',
+    'size_t',
+]
+
+foreach check_header: check_headers
+    has = cxx.has_header(check_header)
+    name = 'HAVE_' + check_header.to_upper().replace('.','_').replace('/','_')
+    conf.set(name, has ? 1 : false, description: 'Define to 1 if you have the <@0@> header file.'.format(check_header))
+endforeach
+
+foreach check_function: check_functions
+    has = cxx.has_function(check_function[0], prefix: check_function[1])
+    name = 'HAVE_' + check_function[0].to_upper().replace('.','_').replace('/','_')
+    conf.set(name, has ? 1 : false, description: 'Define to 1 if you have the `@0@\' function.'.format(check_function[0]))
+endforeach
+
+foreach sizeof_type: sizeof_types
+    size = cxx.sizeof(sizeof_type, prefix : '#include<stdlib.h>')
+    name = 'SIZEOF_' + sizeof_type.to_upper().replace(' ','_')
+    conf.set(name, size, description: 'The size of `@0@\', as computed by sizeof'.format(sizeof_type))
+endforeach
+
+
+
+enable_gcc_atomic_ops_code = '''#include <sched.h>
+int
+main ()
+{
+  int a = 10;
+  __sync_fetch_and_add(&a, 10);
+  __sync_val_compare_and_swap(&a, 0, 10);
+  sched_yield();
+  return 0;
+}
+'''
+enable_gcc_atomic_ops = cxx.compiles(enable_gcc_atomic_ops_code)
+conf.set('HAVE_GCC_ATOMIC_OPS', enable_gcc_atomic_ops ? 1 : false)
+
+enable_osx_atomic_ops_code = '''#include <libkern/OSAtomic.h>
+
+int
+main ()
+{
+  int a = 10;
+  OSAtomicAdd32(10, &a);
+  OSAtomicCompareAndSwapInt(10, 0, &a);
+  return 0;
+}
+'''
+enable_osx_atomic_ops = cxx.compiles(enable_osx_atomic_ops_code)
+conf.set('HAVE_OSX_ATOMIC_OPS', enable_osx_atomic_ops ? 1 : false)
+
+enable_tls_code = '''__thread int a = 0;
+
+int
+main ()
+{
+  a = 10;
+  return 0;
+}
+'''
+enable_tls = cxx.compiles(enable_tls_code)
+conf.set('HAVE_TLS_KEYWORD', enable_tls ? 1 : false)
+
+
+have_unsigned_long_long_int = cxx.sizeof('unsigned long long int', prefix : '#include<stdlib.h>')
+conf.set('HAVE_UNSIGNED_LONG_LONG_INT', have_unsigned_long_long_int != -1 ? 1 : false, description : 'Define to 1 if the system has the type `unsigned long long int\'.')
+
+if cxx.sizeof('off_t', prefix : '#include<sys/types.h>') == -1
+    conf.set('off_t', 'long int', description : 'Define to `long int\' if <sys/types.h> does not define.')
+endif
+
+if cxx.sizeof('size_t', prefix : '#include<sys/types.h>') == -1
+    conf.set('size_t', 'unsigned int', description : 'Define to `unsigned int\' if <sys/types.h> does not define.')
+endif
+
+words_bigendian = host_machine.endian() == 'big'
+conf.set('WORDS_BIGENDIAN', words_bigendian ? 1 : false)
+
+iconvdeps = cxx.find_library('iconv', required: false)
+if iconvdeps.found()
+    deps += iconvdeps
+endif
+
+have_iconv = cxx.has_function('iconv', prefix: '#include <iconv.h>', dependencies: iconvdeps)
+conf.set('HAVE_ICONV', have_iconv, description : 'Define if you have the iconv() function and it works.')
```

Comment:
If you add `meson_version: '>=0.61.0'` to the project() function, then a better way to do this is to rely on the builtin dependency that was added for this: `dependency('iconv', required: false)`. It handles several cases, including one that isn't handled here -- mingw will detect a non-working stub, see https://github.com/mesonbuild/meson/commit/1e19757899fe4388766d71244a2143016ca939ec

Reply:
Thank you for pointing it out.
I'll reflect it in the future release.